Use conda's SQLite instead of the vendored one;
and honor compiler and linker flags.

--- a/Makefile
+++ b/Makefile
@@ -89,9 +89,9 @@
 		-fPIC -shared \
 		-Wall -Wextra \
-		-Ivendor/ \
+		$(CPPFLAGS) \
 		-O3 \
 		$(CFLAGS) \
-		$< -o $@
+		$< -o $@ $(LDFLAGS) $(LDLIBS)
 
 $(TARGET_STATIC): sqlite-vec.c sqlite-vec.h $(prefix) $(OBJS_DIR)
 	$(CC) -Ivendor/ $(CFLAGS) -DSQLITE_CORE -DSQLITE_VEC_STATIC \
